.github/workflows/test-github-user-type.yml

```diff
@@ -0,0 +1,72 @@
+name: test-github-user-type
+
+on:
+  merge_group: ~
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: 'Dry run'
+        required: true
+        default: false
+        type: boolean
+  push:
+    paths:
+      - '.github/workflows/test-github-user-type.yml'
+      - 'github/user-type/**'
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    if: always()
+    needs:
+      - is-bot
+      - is-not-bot
+    runs-on: ubuntu-latest
+    steps:
+      - id: check
+        uses: elastic/oblt-actions/check-dependent-jobs@v1
+        with:
+          jobs: ${{ toJSON(needs) }}
+      - run: ${{ steps.check.outputs.is-success }}
+
+  is-bot:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: elastic/oblt-actions/git/setup@v1
+      - name: Get token
+        id: get_token
+        uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2.1.0
+        with:
+          app_id: ${{ secrets.OBS_AUTOMATION_APP_ID }}
+          private_key: ${{ secrets.OBS_AUTOMATION_APP_PEM }}
```

Comment:
I think we can limit here the permissions, see https://github.com/tibdex/github-app-token#example-workflow.

See https://github.com/elastic/oblt-actions/blob/a9df927b56cafea2460dcc515cbbe7232f359a93/.github/workflows/test-github-is-member-of.yml#L40-L49